uast/transformer/semantic.go

```diff
@@ -161,96 +164,102 @@ func (c *commentElems) isTab(r rune) bool {
 	return false
 }
 
-func (c *commentElems) Split(text string) bool {
-	if c.DoTrim {
-		text = strings.TrimLeftFunc(text, unicode.IsSpace)
+func max(x, y int) int {
+    if x > y {
+        return x
+    }
+    return y
+}
+
+// Returns the first index i of runes satisfying f(runes[i]), len(runes) otherwise
+func firstIndexFunc(runes []rune, f func(r rune) bool) int {
+	for i, r := range runes {
+		if (f(r)) {
+			return i;
+		}
 	}
+	return len(runes);
+}
 
-	if !strings.HasPrefix(text, c.StartToken) || !strings.HasSuffix(text, c.EndToken) {
-		return false
+// Returns the last index i of runes satisfying f(runes[i]), -1 otherwise
+func lastIndexFunc(runes []rune, f func(r rune) bool) int {
+	for i := range runes {
```

Comment:
Check std. lib. 
I think it's better to do:
```
for i := len(s); i > 0; {
		r, size := utf8.DecodeLastRuneInString(s[0:i])
		i -= size
...
```

Reply:
If all of you agree it is better to use the `std` version I can change it, but again this would imply more conversions `string <-> [] rune`